blocksync/blocksync.go

```diff
@@ -231,12 +198,28 @@ func (bs *blockSyncer) ProcessBlock(ctx context.Context, peer string, blk *block
 		return nil
 	}
 
-	bs.buf.AddBlock(bs.tipHeightHandler(), newPeerBlock(peer, blk))
+	tip := bs.tipHeightHandler()
+	if !bs.buf.AddBlock(tip, newPeerBlock(peer, blk)) {
+		return nil
+	}
 	bs.mu.Lock()
 	defer bs.mu.Unlock()
 	if blk.Height() > bs.targetHeight {
 		bs.targetHeight = blk.Height()
 	}
+	syncedHeight := tip
+	for {
+		if !bs.commitBlocks(bs.buf.Delete(syncedHeight + 1)) {
+			break
+		}
+		syncedHeight++
+	}
+	bs.buf.Cleanup(syncedHeight)
+	log.L().Debug("flush blocks", zap.Uint64("start", tip), zap.Uint64("end", syncedHeight))
+	if syncedHeight > bs.lastTip {
+		bs.lastTip = syncedHeight
+		bs.lastTipUpdateTime = time.Now()
+	}
```

Comment:
seems `lastTip` is set here, but never used, i think it should replace `targetHeight`?

for example, we received block 87, 88 first, then `ProcessBlock(block 86)`, as a result, 86, 87, 88 are committed, and got `bs.lastTip = 88` here
but `bs.targetHeight = blk.Height()`, value is 86, shouldn't it be 88 as well?

Reply:
lastTip is used in line 219.
targetHeight is only set if blk.height > targetHeight

Reply:
bs.lastTip only used in:
```
if syncedHeight > bs.lastTip {
		bs.lastTip = syncedHeight
		bs.lastTipUpdateTime = time.Now()
	}
```
so value of `bs.lastTip` being updated, but not used anywhere else in the code

Reply:
instead of 
```
if blk.Height() > bs.targetHeight {
		bs.targetHeight = blk.Height()
	}
```
should use bs.lastTip
```
if bs.lastTip > bs.targetHeight {
		bs.targetHeight = bs.lastTip
	}
```

Reply:
this is exactly what lasttip for